packages/cardano-services/src/Program/utils.ts

```diff
@@ -0,0 +1,236 @@
+/* eslint-disable max-len */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { ClientConfig, Pool, QueryConfig } from 'pg';
+import { HttpServerOptions } from './loadHttpServer';
+import { InMemoryCache, UNLIMITED_CACHE_TTL } from '../InMemoryCache';
+import { InvalidArgsCombination, MissingProgramOption } from './errors';
+import { ProgramOptionDescriptions } from './ProgramOptionDescriptions';
+import { ProviderError, ProviderFailure, TxSubmitProvider } from '@cardano-sdk/core';
+import { RabbitMqTxSubmitProvider } from '@cardano-sdk/rabbitmq';
+import { ServiceNames } from './ServiceNames';
+import { WebSocketClosed, ogmiosTxSubmitProvider, urlToConnectionConfig } from '@cardano-sdk/ogmios';
+import Logger from 'bunyan';
+import dns, { SrvRecord } from 'dns';
+import pRetry, { FailedAttemptError } from 'p-retry';
+
+export const SERVICE_DISCOVERY_BACKOFF_FACTOR_DEFAULT = 1.1;
+export const SERVICE_DISCOVERY_BACKOFF_TIMEOUT_DEFAULT = 60 * 1000;
+export const DNS_SRV_CACHE_KEY = 'DNS_SRV_Record';
+
+export type RetryBackoffConfig = {
+  factor?: number;
+  maxRetryTime?: number;
+};
+
+export const onFailedAttemptFor =
+  (serviceName: string, logger: Logger) =>
+  async ({ attemptNumber, message, retriesLeft }: FailedAttemptError) => {
+    const nextAction = retriesLeft > 0 ? 'retrying...' : 'exiting';
+    logger.trace(message);
+    logger.debug(
+      `Establishing connection to ${serviceName}: Attempt ${attemptNumber} of ${
+        attemptNumber + retriesLeft
+      }, ${nextAction}`
+    );
+    if (retriesLeft === 0) {
+      logger.error(message);
+      // Invokes onDeath() callback within cardano-services entrypoints, following by server.shutdown() and process.exit(1)
+      process.kill(process.pid, 'SIGTERM');
+    }
+  };
+
+// Get a random selection of dns srv resolved service address initially and make it sticky for reconnects by storing a reference in memory
+export const getStikyAddressWithDnsSrv = async (serviceName: string, cache: InMemoryCache): Promise<SrvRecord> => {
```

Comment:
```suggestion
export const resolveMaybeCachedSrvRecord = async (serviceName: string, cache: InMemoryCache): Promise<SrvRecord> => {
```
I think we can probably depend on the concept of a cached record, rather than use the term _sticky_.